src/CoreWCF.UnixDomainSocket/src/CoreWCF/Channels/UnixPosixIdentityBindingElement.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.Security;
+using CoreWCF.Security;
+
+namespace CoreWCF.Channels
+{
+    public class UnixPosixIdentityBindingElement : StreamUpgradeBindingElement
+    {
+        private ProtectionLevel _protectionLevel;
+
+        public UnixPosixIdentityBindingElement()
+            : base()
+        {
+            _protectionLevel = ConnectionOrientedTransportDefaults.ProtectionLevel;
+        }
+
+        protected UnixPosixIdentityBindingElement(UnixPosixIdentityBindingElement elementToBeCloned)
+            : base(elementToBeCloned)
+        {
+            _protectionLevel = elementToBeCloned._protectionLevel;
+        }
+
+        public ProtectionLevel ProtectionLevel
+        {
+            get
+            {
+                return _protectionLevel;
+            }
+            set
+            {
+                ProtectionLevelHelper.Validate(value);
+                _protectionLevel = value;
+            }
+        }
+
+        public override BindingElement Clone()
+        {
+            return new UnixPosixIdentityBindingElement(this);
+        }
+
+        public override StreamUpgradeProvider BuildServerStreamUpgradeProvider(BindingContext context)
+        {
+            return new UnixPosixIdentitySecurityUpgradeProvider(this, context);
+        }
```

Comment:
You can get the binding from the BindingContext. Can you check the scheme to make sure it's net.uds so we don't have to worry about someone trying to use this for named pipes or net.tcp.